runtime/lua/vim/lsp/completion.lua

```diff
@@ -0,0 +1,11 @@
+local M = {}
+
+--- @alias vim.lsp.CompletionResult lsp.CompletionList | lsp.CompletionItem[]
```

Comment:
Why the separate file for this? Can this go into _completion?

If we're going to have more public API I think we can then rename that module. Having both _completion and completion shouldn't be necessary(?)

Reply:
It's because the work of #27542 will be done in `completion.lua`, and because I feel these are "public" types it made sense to put it here, unless we want to initially put all of the completion work in the private module.

Reply:
> unless we want to initially put all of the completion work in the private module.

Either that, or we already rename `_completion` to `completion.lua`.
Not sure how we deal with the omnifunc, because it is also exposed under `vim.lsp.omnifunc`. Could rename it to `_omnifunc`, or deprecate `vim.lsp.omnifunc` in favour of `vim.lsp.completion.omnifunc`. No strong opinion on that.

What I'd like to avoid is having a public `completion.lua` that calls into a private `_completion.lua`.

Reply:
Alright I'll move this to `_completion.lua` (we might rename it in #27542). The omnifunc issue might be a larger/longer discussion.